src/Components/App/App.js

```diff
@@ -5,22 +5,72 @@ import Header from '../Header/Header';
 import Home from '../Home/Home';
 import Horoscope from '../Horoscope/Horoscope';
 import Footer from '../Footer/Footer';
+import { fetchHoroscope } from '../../apiCalls';
 
 class App extends Component {
   constructor() {
     super() 
     this.state = {
+      isLoading: true,
+      errorMsg: '',
+      day: '',
+      sign: '',
+      horoscope: '',
+      image: ''
+    }
+  }
+
+  retrieveHoroscopeData = async (url, src, alt, when) => {
+    await fetchHoroscope(url)
+    .then(result => {
+        this.setState({ isLoading: true, horoscope: '' })
+        if (typeof result === 'string') {
+            setTimeout(() => {
+                this.setState({
+                    isLoading: false,
+                    errorMsg: result
+                })
+            }, 600)
+          } else {
+            setTimeout(() => {
+                this.setState({
+                    day: when || 'today',
+                    image: src,
+                    sign: alt, 
+                    horoscope: result,
+                    isLoading: false
+                })
+            }, 600)   
+          }
+    })
+  }
 
+  retrieveDifferentDay = (when) => {
+    let url = `https://aztro.sameerkumar.website/?sign=${this.state.sign}&day=${when}`;
```

Comment:
Is there a way you could store the various urls in the apiCalls file as variables and dynamically access them through what you pass into the api call? might clean this up a little!